### PR TITLE
Added nightly processes and updated them for the new OGC API Proc definition

### PR DESCRIPTION
**Pull request by RousseauLambertLP**
* Added `msc-pygeoapi-config-nightly.yml ` for nightly only process. This way, eventually we<ll be able to cat this nightly config file with the `deploy/default/msc-pygeoapi-config.yml` file for nightly.
*  Added `MSC_PYGEOAPI_BASEPATH` in the env variable file. Previously it was set using a relative path which caused problem when building in conda for example. Now it is set in the env file.
* Added the search function to the Elasticsearch loader, this was needed for vigilance and extract raster processes.
* Updated the extract raster process for:
    * Added an example input for OpenAPI
    * Changed to use the elasticsearch loader and configuration of msc-pygeoapi
    * Updated the pyproj function for the pyproj version we have in gemet-dev-03
* Reworked the vigilance process created last year to update it.

CC @tom-cooney

### CODE REVIEW

File: deploy/nightly/msc-pygeoapi-config-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+    Extract-Raster:
```
Comment (tomkralidis) [inline deploy/nightly/msc-pygeoapi-config-nightly.yml:1]: `extract-raster`

File: deploy/nightly/msc-pygeoapi-config-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+        type: process
+        processor:
+            name: msc_pygeoapi.process.weather.extract_raster.ExtractRasterProcessor
+    Generate-Vigilance:
```
Comment (tomkralidis) [inline deploy/nightly/msc-pygeoapi-config-nightly.yml:5]: `generate-vigilance`

File: msc-pygeoapi.env [hunk 3]
```diff
@@ -15,6 +15,7 @@ export MSC_PYGEOAPI_STATIC=theme/static
 
 export XDG_CACHE_HOME=/tmp/msc-pygeoapi-sarra-logs
 
+export MSC_PYGEOAPI_BASEPATH=/opt/msc-pygeoapi/
```
Comment (tomkralidis) [inline msc-pygeoapi.env:18]: We shouldn't need to set this.

File: msc_pygeoapi/connector/elasticsearch_.py [hunk 4]
```diff
@@ -276,9 +276,24 @@ def update_by_query(self, query, name):
         :return: `bool` of index update status
         """
 
-        self.Elasticsearch.update_by_query(body=query, index=name)
+        try:
+            self.Elasticsearch.update_by_query(body=query, index=name)
+        except Exception:
```
Comment (tomkralidis) [inline msc_pygeoapi/connector/elasticsearch_.py:281]: Let's make the exception more explicit and log it.

File: msc_pygeoapi/env.py [hunk 5]
```diff
@@ -59,4 +59,4 @@
 else:
     MSC_PYGEOAPI_ES_AUTH = (MSC_PYGEOAPI_ES_USERNAME, MSC_PYGEOAPI_ES_PASSWORD)
 
-MSC_PYGEOAPI_BASEPATH = os.path.dirname(os.path.realpath(__file__))
+MSC_PYGEOAPI_BASEPATH = os.getenv('MSC_PYGEOAPI_BASEPATH', None)
```
Comment (tomkralidis) [inline msc_pygeoapi/env.py:62]: let's discuss.

**Review by tomkralidis — CHANGES_REQUESTED** [review]
Good work here @RousseauLambertLP !  See comments.  As well, rename `msc_pygeoapi/resources/images/logo.png` to `msc_pygeoapi/resources/images/eccc-logo.png`

File: deploy/nightly/msc-pygeoapi-config-nightly.yml (hunk 1 above)
Comment (RousseauLambertLP) [reply deploy/nightly/msc-pygeoapi-config-nightly.yml]: fixed

File: deploy/nightly/msc-pygeoapi-config-nightly.yml (hunk 2 above)
Comment (RousseauLambertLP) [reply deploy/nightly/msc-pygeoapi-config-nightly.yml]: fixed

File: msc_pygeoapi/connector/elasticsearch_.py (hunk 4 above)
Comment (RousseauLambertLP) [reply msc_pygeoapi/connector/elasticsearch_.py]: I have updated it to : 
```
        try:
            self.Elasticsearch.update_by_query(body=query, index=name)
        except  exceptions.NotFoundError as err:
            LOGGER.error('Could not update index: {}'.format(err))
            return False
```

**Comment by RousseauLambertLP** [comment]
Closing, as these process will be revisited and updated eventually.

**Pull request closed without merging**